### PR TITLE
Remove allowance in case of failed swap transaction

### PR DESCRIPTION
**Pull request by Dubiy**

### CODE REVIEW

File: src/services/CryptoFlowService.ts [hunk 1]
```diff
@@ -54,11 +54,13 @@ export const executeAllowanceApprove = async(selectedQuote: ICryptoFlowQuote, ac
...
-            Math.pow(10, sourceToken!.decimals!) * (
-                selectedQuote.source.amountExpected! * 1.2 // with a little buffer
+        const amount = approve
+            ? Math.ceil(
+                Math.pow(10, sourceToken!.decimals!) * (
+                    selectedQuote.source.amountExpected! * 1.2 // with a little buffer
+                )
             )
-        );
+            : 0;
```
Comment (Volind) [inline src/services/CryptoFlowService.ts:63]: Here could be an issue if user already has an allowance for this token in place

Comment (Dubiy) [reply src/services/CryptoFlowService.ts:63]: we decided to skip for now

File: src/BladeSDK.ts [hunk 2]
```diff
@@ -626,7 +627,7 @@ export class BladeSDK {
      */
     async getKeysFromMnemonic(mnemonicRaw: string, lookupNames: boolean, completionKey?: string): Promise<PrivateKeyData> {
         try {
-            const mnemonic = await Mnemonic.fromString(mnemonicRaw);
+            const mnemonic = await Mnemonic.fromString(mnemonicRaw.toLowerCase());
```
Comment (Volind) [inline src/BladeSDK.ts]: if it's a plain user input, shouldn't we also trim it?

Comment (Dubiy) [reply src/BladeSDK.ts]:
```
"         dddd bbb       cccc    ".split(" ").filter(word => word).join(" ")
```

`// 'dddd bbb cccc'`

File: src/BladeSDK.ts [hunk 3]
```diff
@@ -626,7 +627,12 @@ export class BladeSDK {
      */
     async getKeysFromMnemonic(mnemonicRaw: string, lookupNames: boolean, completionKey?: string): Promise<PrivateKeyData> {
         try {
-            const mnemonic = await Mnemonic.fromString(mnemonicRaw);
+            const mnemonic = await Mnemonic.fromString(mnemonicRaw
+                .toLowerCase()
+                .split(" ")
+                .filter(word => word)
```
Comment (Volind) [inline src/BladeSDK.ts:633]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim

**Review by Volind — CHANGES_REQUESTED** [review]

File: src/BladeSDK.ts (hunk 3 above)
Comment (Dubiy) [reply src/BladeSDK.ts:633]: trim not help if there will be 2 spaces in row between words. this solution helps

**Review by Volind — APPROVED** [review]

**Pull request merged**